src/plugins/intel_npu/CMakeLists.txt

```diff
@@ -26,6 +26,46 @@ endif()
 
 add_subdirectory(thirdparty EXCLUDE_FROM_ALL)
 
+#
+# Warnings as errors
+#
+
+macro(remove_compiler_flags FLAGS)
+    string(REPLACE " " ";" _tokened_list "${FLAGS}")
+    foreach(flag IN LISTS _tokened_list)
+        string(REPLACE "${flag}" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+        string(REPLACE "${flag}" "" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
+        string(REPLACE "${flag}" "" CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
+        string(REPLACE "${flag}" "" CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
+    endforeach()
+endmacro()
+
+if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+    set(CMAKE_COMPILE_WARNING_AS_ERROR ON)
+    remove_compiler_flags(/wd4996)
+    if(CMAKE_VERSION VERSION_LESS 3.24)
+        ov_add_compiler_flags(/WX-)
```

Comment:
I understand `/WX-` would be the opposite of `/WX` and it means that we should NOT treat warnings as errors for CMake versions less than 3.24. Is this something that we want to do?

Reply:
Good catch. I intended to use `/WX`, but that must've slipped through.